trading_cost.py

```diff
@@ -0,0 +1,64 @@
+"""
+A file to build cost objects.
+"""
+
+from collections import defaultdict
+
+class CurrencyCost:
+    """
+    Cost function of currency trading.
+    """
+    def __init__(self, base_ins, instruments):
+        self.base_ins = base_ins
+        self.instruments = instruments
+        self.quantity = defaultdict(int)
+        self.value = defaultdict(int)
+        self.value[base_ins] = 1
+        self.quantity[base_ins] = 1
+
+    def __call__(self, portfolio, price, liquidation):
+        if liquidation:
+            for instrument in self.instruments:
+                portfolio[-1][instrument] = (
+                    0 if instrument is not self.base_ins else 1)
+        return self.calculate(portfolio[-1], price)
+
+    def calculate(self, today, price):
+        """
+        Calculate pnl and produce cost.
+
+        We assume that our investing money is maintained in 1.
+        That make it more general to evaluate the perfoormance for the alpha.
+        For detailed, please read our wiki pages.
+        """
+        pnl = {
+            instrument : 0 for instrument in self.instruments}
+        cost = {
+            instrument : 0 for instrument in self.instruments}
+        total_value = 0
+        for instrument in today:
+            if instrument not in self.instruments:
+                raise KeyError("%s is not tradable." % instrument)
+            if (price[instrument][-1].sell is not None and
+                    price[instrument][-1].sell is not None):
+                position = today[instrument]
+                middle_price = (
+                    (price[instrument][-1].sell +
+                     price[instrument][-1].sell) / 2)
+                difference = (
+                    self.quantity[instrument] - position / middle_price)
+                spread = (
+                    (price[instrument][-1].sell -
+                     price[instrument][-1].buy) / 2)
+                self.quantity[instrument] = position / middle_price
+                self.value[instrument] = position
+                cost[instrument] = abs(difference * spread)
+                pnl[instrument] = (
+                    difference * middle_price - cost[instrument])
+        total_value = sum(
+            self.value[instrument] for instrument in self.instruments)
+        short_value = 1 - total_value
+        pnl[self.base_ins] -= short_value
```

Comment:
Base instrument, whatever it is, should already be in portfolio.
And that's the only way users can hold base instrument.
Indeed, we will run Portfolio.normailze() before calculating anything here.
So it is impossible that total_value is not 1.

Reply:
Oh! It's possible that total_value is not 1 due to line 42.
We may assign portfolio for every instrument (included base instrument), but not every instrument would be traded. ex.  
10/01  TWD  0, USD 0.4, JPY 0.6
10/02  TWD  0, USD 0.5, JPY 0.5
But JPY couldn't trade on 10/02, so the JPY account will remain 0.6 (Base on our last discussion.) . That the total_value will be 1.1, we should do this line to adjust the value in base instrument.

Reply:
I see. I forget such cases, and that's definitely correct. Thanks!